### PR TITLE
Add BrightnessValue ring algebra

### PR DESCRIPTION
**Pull request by cquiroz**
I wanted to add some basic operations on brightness value but then I thought I could directly make a Ring. Maybe overkill?

### CODE REVIEW

**Review by swalker2m — APPROVED** [review]

File: modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala [hunk 1]
```diff
@@ -86,4 +87,31 @@ object BrightnessValue {
...
+      override def negate(a: BrightnessValue): BrightnessValue =
+        new BrightnessValue(-a.scaledValue)
+
+      override val zero: BrightnessValue =
+        new BrightnessValue(0)
+
+      override val one: BrightnessValue =
+        new BrightnessValue(1)
+
+      override def times(x: BrightnessValue, y: BrightnessValue): BrightnessValue =
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala:105]: I originally did just an `AdditiveGroup` which enables addition and subtraction,  but a Ring enables to do scalar multiplication besides regular multiplication.

I don't think we'll multiply values but scalar multiplication seems useful

[the remark this answers is not in the document]
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala:105]: > I don't think we'll multiply values but scalar multiplication seems useful

Agreed, but the `Ring` defines multiplication with the same type (`BrightnessValue`), not a scalar. It happens that the `BrightnessValue` wraps a unitless scala, but defining multiplication between `BrightnessValue`s to achieve scalar multiplication feels a bit like a backdoor to the type system.

I actually wonder if we can unify `BrightnessValue` (or replace it) with `lucuma-ui`'s  `TruncatedBigDecimal` ?

[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala:105]: Right, maybe I should go back to the `AdditiveGroup`

[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala:105]: OTOH the fact that we never do multiplication doesn't mean it is not allowed

[the remark this answers is not in the document]
Comment (rpiaggio) [reply modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala]: No strong feelings either.

[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/BrightnessValue.scala]: Ok, let's keep the `AdditiveGroup` which we'll certainly use

**Pull request merged**